connectivity/FEATURE_BLE/source/cordio/source/GattServerImpl.cpp

```diff
@@ -989,9 +1013,10 @@ uint8_t GattServer::atts_read_cb(
             AUTH_CALLBACK_REPLY_SUCCESS
         };
 
-        GattAuthCallbackReply_t ret = auth_char->authorizeRead(&read_auth_params);
-        if (ret != AUTH_CALLBACK_REPLY_SUCCESS) {
-            return ret & 0xFF;
+        auth_cb->read_cb.call(&read_auth_params);
```

Comment:
I don't know what `FunctionPointerWithContext` does, but the test above followed by this by-value access to the member is a bit confusing.